pkg/ingester/kafka_consumer.go

```diff
@@ -107,6 +109,7 @@ func (kc *kafkaConsumer) consume(records []partition.Record) {
 		}); err != nil {
 			level.Error(kc.logger).Log("msg", "failed to push records", "error", err)
 		}
+		kc.committer.EnqueueOffset(record.Offset)
```

Comment:
Are the records sorted by offset? I'm assuming warpstream returns them that way but I don't think we have a guarantee around that?
If they're out of order, we might commit the wrong offset. Not a massive deal as it would be corrected on the next batch, but it would cause some duplications if the ingester crashed/restarted right after.

Reply:
They are sorted yes. I've also considered doing it once for the whole batch.

Reply:
This reminds me of a different problem. What happens if one of the records fails but the next one succeeds?

Reply:
It's possible as we test this out we realize there are ones that needs to be retried specially during shutdown and we'll change that code to catch them but right now it's hard to figure this.